From 69417017fb071c763db1bcca360d85fbe60c9139 Mon Sep 17 00:00:00 2001
From: John Linn <john.linn@xilinx.com>
Date: Tue, 27 Nov 2012 16:11:19 -0800
Subject: [PATCH] arm: zynq: Add support for Zed Board

Basic testing has been done with ethernet, usb host,
QSPI (16MB) and SD. The device tree was generated using the
merged device tree generator.

The QSPI partitions are setup assuming a new memory map
that is going to be supported in u-boot also.

Signed-off-by: John Linn <john.linn@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/boot/dts/zynq-zed.dts   |  311 ++++++++++++++++++++++++++++++++++++++
 arch/arm/mach-zynq/Makefile.boot |    3 +-
 2 files changed, 313 insertions(+), 1 deletions(-)
 create mode 100644 arch/arm/boot/dts/zynq-zed.dts

diff --git a/arch/arm/boot/dts/zynq-zed.dts b/arch/arm/boot/dts/zynq-zed.dts
new file mode 100644
index 0000000..bc9f168
--- /dev/null
+++ b/arch/arm/boot/dts/zynq-zed.dts
@@ -0,0 +1,311 @@
+/*
+ * Device Tree Generator version: 1.1
+ *
+ * (C) Copyright 2007-2012 Xilinx, Inc.
+ * (C) Copyright 2007-2012 Michal Simek
+ * (C) Copyright 2007-2012 PetaLogix Qld Pty Ltd
+ *
+ * Michal SIMEK <monstr@monstr.eu>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ *
+ * CAUTION: This file is automatically generated by libgen.
+ * Version: Xilinx EDK 14.3 EDK_P.40xd
+ * Today is: Tuesday, the 27 of November, 2012; 12:29:21
+ *
+ * XPS project directory: device-tree_bsp_0
+ */
+
+/dts-v1/;
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	compatible = "xlnx,zynq-zc770";
+	model = "Xilinx Zynq";
+	aliases {
+		ethernet0 = &ps7_ethernet_0;
+		serial0 = &ps7_uart_1;
+	} ;
+	chosen {
+		bootargs = "console=ttyPS0,115200 root=/dev/ram rw ip=:::::eth0:dhcp earlyprintk";
+		linux,stdout-path = "/axi@0/serial@e0001000";
+	} ;
+	cpus {
+		#address-cells = <1>;
+		#cpus = <0x2>;
+		#size-cells = <0>;
+		ps7_cortexa9_0: cpu@0 {
+			clock-frequency = <666666687>;
+			compatible = "xlnx,ps7-cortexa9-1.00.a";
+			d-cache-line-size = <0x20>;
+			d-cache-size = <0x8000>;
+			device_type = "cpu";
+			i-cache-line-size = <0x20>;
+			i-cache-size = <0x8000>;
+			model = "ps7_cortexa9,1.00.a";
+			reg = <0>;
+			timebase-frequency = <333333343>;
+			xlnx,cpu-1x-clk-freq-hz = <0x69f6bcb>;
+			xlnx,cpu-clk-freq-hz = <0x27bc86bf>;
+		} ;
+		ps7_cortexa9_1: cpu@1 {
+			clock-frequency = <666666687>;
+			compatible = "xlnx,ps7-cortexa9-1.00.a";
+			d-cache-line-size = <0x20>;
+			d-cache-size = <0x8000>;
+			device_type = "cpu";
+			i-cache-line-size = <0x20>;
+			i-cache-size = <0x8000>;
+			model = "ps7_cortexa9,1.00.a";
+			reg = <1>;
+			timebase-frequency = <333333343>;
+			xlnx,cpu-1x-clk-freq-hz = <0x69f6bcb>;
+			xlnx,cpu-clk-freq-hz = <0x27bc86bf>;
+		} ;
+	} ;
+	ps7_ddr_0: memory@0 {
+		device_type = "memory";
+		reg = < 0x0 0x20000000 >;
+	} ;
+	ps7_axi_interconnect_0: axi@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "xlnx,ps7-axi-interconnect-1.00.a", "simple-bus";
+		ranges ;
+		ps7_afi_0: ps7-afi@f8008000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = < 0xf8008000 0x1000 >;
+		} ;
+		ps7_afi_1: ps7-afi@f8009000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = < 0xf8009000 0x1000 >;
+		} ;
+		ps7_afi_2: ps7-afi@f800a000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = < 0xf800a000 0x1000 >;
+		} ;
+		ps7_afi_3: ps7-afi@f800b000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = < 0xf800b000 0x1000 >;
+		} ;
+		ps7_ddrc_0: ps7-ddrc@f8006000 {
+			compatible = "xlnx,ps7-ddrc-1.00.a";
+			reg = < 0xf8006000 0x1000 >;
+			xlnx,has-ecc = <0x0>;
+		} ;
+		ps7_dev_cfg_0: ps7-dev-cfg@f8007000 {
+			compatible = "xlnx,ps7-dev-cfg-1.00.a";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 8 0 >;
+			reg = < 0xf8007000 0x1000 >;
+		} ;
+		ps7_dma_ns: ps7-dma@f8004000 {
+			compatible = "xlnx,ps7-dma-1.00.a";
+			reg = < 0xf8004000 0x1000 >;
+		} ;
+		ps7_dma_s: ps7-dma@f8003000 {
+			compatible = "xlnx,ps7-dma-1.00.a";
+			reg = < 0xf8003000 0x1000 >;
+		} ;
+		ps7_ethernet_0: ps7-ethernet@e000b000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "xlnx,ps7-ethernet-1.00.a";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 22 0 >;
+			local-mac-address = [ 00 0a 35 00 00 00 ];
+			phy-handle = <&phy0>;
+			reg = < 0xe000b000 0x1000 >;
+			xlnx,enet-clk-freq-hz = <0x7735940>;
+			xlnx,enet-reset = <0xffffffff>;
+			xlnx,enet-slcr-1000mbps-div0 = <0x8>;
+			xlnx,enet-slcr-1000mbps-div1 = <0x1>;
+			xlnx,enet-slcr-100mbps-div0 = <0x8>;
+			xlnx,enet-slcr-100mbps-div1 = <0x5>;
+			xlnx,enet-slcr-10mbps-div0 = <0x8>;
+			xlnx,enet-slcr-10mbps-div1 = <0x32>;
+			xlnx,eth-mode = <0x1>;
+			xlnx,has-mdio = <0x1>;
+			xlnx,ptp-enet-clock = <111111115>;
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy0: phy@7 {
+					compatible = "marvell,88e1510";
+					device_type = "ethernet-phy";
+					reg = <7>;
+				} ;
+			} ;
+		} ;
+		ps7_gpio_0: ps7-gpio@e000a000 {
+			#gpio-cells = <2>;
+			compatible = "xlnx,ps7-gpio-1.00.a";
+			emio-gpio-width = <64>;
+			gpio-controller ;
+			gpio-mask-high = <0xc0000>;
+			gpio-mask-low = <0xfe81>;
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 20 0 >;
+			reg = < 0xe000a000 0x1000 >;
+		} ;
+		ps7_iop_bus_config_0: ps7-iop-bus-config@e0200000 {
+			compatible = "xlnx,ps7-iop-bus-config-1.00.a";
+			reg = < 0xe0200000 0x1000 >;
+		} ;
+		ps7_pl310_0: ps7-pl310@f8f02000 {
+			arm,data-latency = < 3 2 2 >;
+			arm,tag-latency = < 2 2 2 >;
+			cache-level = < 2 >;
+			cache-unified ;
+			compatible = "xlnx,ps7-pl310-1.00.a", "arm,pl310-cache";
+			reg = < 0xf8f02000 0x1000 >;
+		} ;
+		ps7_qspi_0: ps7-qspi@e000d000 {
+			bus-num = <0>;
+			compatible = "xlnx,ps7-qspi-1.00.a";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 19 0 >;
+			is-dual = <0>;
+			num-chip-select = <1>;
+			reg = < 0xe000d000 0x1000 >;
+			speed-hz = <200000000>;
+			xlnx,fb-clk = <0x1>;
+			xlnx,qspi-clk-freq-hz = <0xbebc200>;
+			xlnx,qspi-mode = <0x0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			flash@0 {
+				compatible = "n25q128";
+				reg = <0x0>;
+				spi-max-frequency = <50000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				partition@qspi-fsbl-uboot {
+					label = "qspi-fsbl-uboot";
+					reg = <0x0 0x100000>;
+				};
+				partition@qspi-linux {
+					label = "qspi-linux";
+					reg = <0x100000 0x500000>;
+				};
+				partition@qspi-device-tree {
+					label = "qspi-device-tree";
+					reg = <0x600000 0x20000>;
+				};
+				partition@qspi-rootfs {
+					label = "qspi-rootfs";
+					reg = <0x620000 0x9D0000>;
+				};
+			};
+		} ;
+		ps7_qspi_linear_0: ps7-qspi-linear@fc000000 {
+			compatible = "xlnx,ps7-qspi-linear-1.00.a";
+			reg = < 0xfc000000 0x1000000 >;
+			xlnx,qspi-clk-freq-hz = <0xe4e1c0>;
+		} ;
+		ps7_ram_0: ps7-ram@0 {
+			compatible = "xlnx,ps7-ram-1.00.a";
+			reg = < 0x0 0x30000 >;
+		} ;
+		ps7_ram_1: ps7-ram@ffff0000 {
+			compatible = "xlnx,ps7-ram-1.00.a";
+			reg = < 0xffff0000 0xfe00 >;
+		} ;
+		ps7_scugic_0: ps7-scugic@f8f01000 {
+			#address-cells = < 2 >;
+			#interrupt-cells = < 3 >;
+			#size-cells = < 1 >;
+			compatible = "xlnx,ps7-scugic-1.00.a", "arm,cortex-a9-gic", "arm,gic";
+			interrupt-controller ;
+			linux,phandle = < 0x1 >;
+			phandle = < 0x1 >;
+			reg = < 0xf8f01000 0x1000 0xf8f00100 0x100 >;
+		} ;
+		ps7_scutimer_0: ps7-scutimer@f8f00600 {
+			clock-frequency = <333333343>;
+			compatible = "xlnx,ps7-scutimer-1.00.a", "arm,cortex-a9-twd-timer";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 1 13 0 >;
+			reg = < 0xf8f00600 0x20 >;
+		} ;
+		ps7_scuwdt_0: ps7-scuwdt@f8f00620 {
+			clock-frequency = <333333343>;
+			compatible = "xlnx,ps7-scuwdt-1.00.a";
+			device_type = "watchdog";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 1 14 0 >;
+			reg = < 0xf8f00620 0xe0 >;
+		} ;
+		ps7_sd_0: ps7-sdio@e0100000 {
+			clock-frequency = <50000000>;
+			compatible = "xlnx,ps7-sdio-1.00.a", "generic-sdhci";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 24 0 >;
+			reg = < 0xe0100000 0x1000 >;
+			xlnx,has-cd = <0x1>;
+			xlnx,has-power = <0x0>;
+			xlnx,has-wp = <0x1>;
+			xlnx,sdio-clk-freq-hz = <0x2faf080>;
+		} ;
+		ps7_slcr_0: ps7-slcr@f8000000 {
+			compatible = "xlnx,ps7-slcr-1.00.a";
+			reg = < 0xf8000000 0x1000 >;
+		} ;
+		ps7_ttc_0: ps7-ttc@f8001000 {
+			clock-frequency = <111111115>;
+			clock-frequency-timer0 = <111111115>;
+			clock-frequency-timer1 = <111111115>;
+			clock-frequency-timer2 = <111111115>;
+			compatible = "xlnx,ps7-ttc-1.00.a";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 10 0 0 11 0 0 12 0 >;
+			reg = < 0xf8001000 0x1000 >;
+			xlnx,ttc-clk0-clksrc = <0x0>;
+			xlnx,ttc-clk0-freq-hz = <0x69f6bcb>;
+			xlnx,ttc-clk1-clksrc = <0x0>;
+			xlnx,ttc-clk1-freq-hz = <0x69f6bcb>;
+			xlnx,ttc-clk2-clksrc = <0x0>;
+			xlnx,ttc-clk2-freq-hz = <0x69f6bcb>;
+		} ;
+		ps7_uart_1: serial@e0001000 {
+			clock = <50000000>;
+			clock-frequency = <50000000>;
+			compatible = "xlnx,ps7-uart-1.00.a", "xlnx,xuartps";
+			current-speed = <115200>;
+			device_type = "serial";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 50 0 >;
+			port-number = <0>;
+			reg = < 0xe0001000 0x1000 >;
+			xlnx,has-modem = <0x0>;
+			xlnx,uart-clk-freq-hz = <0x2faf080>;
+		} ;
+		ps7_usb_0: ps7-usb@e0002000 {
+			compatible = "xlnx,ps7-usb-1.00.a";
+			dr_mode = "host";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 21 0 >;
+			phy_type = "ulpi";
+			reg = < 0xe0002000 0x1000 >;
+			xlnx,usb-reset = <0xffffffff>;
+		} ;
+		ps7_xadc: ps7_xadc@f8007100 {
+			compatible = "xlnx,ps7-xadc-1.00.a";
+			interrupts = < 0 7 4 >;
+			reg = < 0xf8007100 0x20 >;
+		} ;
+	} ;
+} ;
diff --git a/arch/arm/mach-zynq/Makefile.boot b/arch/arm/mach-zynq/Makefile.boot
index 9f4921a..7a4faf6 100644
--- a/arch/arm/mach-zynq/Makefile.boot
+++ b/arch/arm/mach-zynq/Makefile.boot
@@ -11,4 +11,5 @@ dtb-$(CONFIG_ARCH_ZYNQ) += \
 			zynq-zc770-xm010.dtb \
 			zynq-zc770-xm011.dtb \
 			zynq-zc770-xm012.dtb \
-			zynq-zc770-xm013.dtb
+			zynq-zc770-xm013.dtb \
+			zynq-zed.dtb
-- 
1.7.1

